teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,471 @@
+  * Must be one of `proposed`, `implementable`,
+    `implemented`,`withdrawn`, or `replaced`.
+* **authors** Required
+  * A list of authors for the TEP. This is simply the github ID. In
+    the future we may enhance this to include other types of
+    identification.
+* **creation-date** Required
+  * The date that the TEP was first submitted in a PR.
+  * In the form `yyyy-mm-dd`
+  * While this info will also be in source control, it is helpful to
+    have the set of TEP files stand on their own.
+* **last-updated** Optional
+  * The date that the TEP was last changed significantly.
+  * In the form `yyyy-mm-dd`
+* **see-also** Optional
+  * A list of other TEPs that are relevant to this TEP.
+  * In the form `TEP-123`
+* **replaces** Optional
+  * A list of TEPs that this TEP replaces. Those TEPs should list
+    this TEP in their `superseded-by`.
+  * In the form `TEP-123`
+* **superseded-by** Optional
+  * A list of TEPs that supersede this TEP. Use of this should be
+    paired with this TEP moving into the `Replaced` status.
+  * In the form `TEP-123`
+
+
+### TEP Workflow
+
+A TEP has the following states
+
+- `proposed`: The TEP has been proposed and is actively being
+  defined. This is the starting state while the TEP is being fleshed
+  out and actively defined and discussed.
+- `implementable`: The approvers have approved this TEP for
+  implementation.
+- `implemented`: The TEP has been implemented and is no longer
+  actively changed. From that point on, the TEP should be considered
+  *read-only*.
+- `withdrawn`: The TEP has been withdrawn by the authors or by the
+  community on agreement with the authors.
+- `replaced`: The TEP has been replaced by a new TEP. The
+  `superseded-by` metadata value should point to the new TEP.
+
+When a TEP is merged with the `proposed` state, this means the
+project owners acknowledge this is something we need to work on *but*
+the proposal needs to be more detailed before we can go ahead and
+implement it in the main project(s). This state doesn't prevent using
+`tektoncd/experimental` to *experiment* and gather feedback.
+
+A TEP can be created with the `implementable` state if it doesn't need
+any more discussion and got approved as it.
+
+See [Examples](#examples) to see examples of TEP workflow on use cases.
+
+### Git Implementation
+
+TEPs are checked into the community repo under the `/teps` directory.
+
+New TEPs can be checked in with a file name in the form of
+`draft-YYYYMMDD-my-title.md`. As significant work is done on the TEP,
+the authors can assign a TEP number. No other changes should be put in
+that PR so that it can be approved quickly and minimize merge
+conflicts. The TEP number can also be done as part of the initial
+submission if the PR is likely to be uncontested and merged quickly.
+
+### Prior Art
+
+The TEP process as proposed was essentially adapted from the
+[Kubernetes KEP process][], which itself is essentially stolen from the [Rust RFC
```

Comment:
Does this process apply to all Tekton projects in addition to Pipelines like CLI, triggers, dashboard? if so, should we store TEPs in a project specific directories like KEP?

Reply:
It applies to all tekton project, but so far, trying to keep it simple and have it all on the same folder. 
1. We may have a lot of TEP that touch more than one project
2. We are not as big as kubernetes :stuck_out_tongue_closed_eyes: 

We'll see over time if we need to have subfolders, etc..